### PR TITLE
fix(cli): do not extend injected script on same-document navigations

### LINKED ISSUE #4993: Two recorders are created in one document under CLI
```sh
node lib/cli/cli.js codegen github.com/microsoft
```
type 'vscode'
click vscode

### PR DESCRIPTION
**Pull request by dgozman**
Otherwise, the injected script has to be ready for reentrancy.

Fixes #4993.

### CODE REVIEW

File: src/server/frames.ts [hunk 1]
```diff
@@ -213,7 +213,6 @@ export class FrameManager {
     const navigationEvent: NavigationEvent = { url, name: frame._name };
     frame.emit(Frame.Events.Navigation, navigationEvent);
     debugLogger.log('api', `  navigated to "${url}"`);
-    this._page.frameNavigated(frame);
```
Comment (pavelfeldman) [inline src/server/frames.ts:216]: Are we no longer reporting frame navigated on the Page object for history API?

File: src/cli/injected/recorder.ts [hunk 2]
```diff
@@ -372,7 +374,7 @@ export class Recorder {
     if (event.key === '@' && event.code === 'KeyL')
       return false;
     // Allow and ignore common used shortcut for pasting.
-    if (process.platform === 'darwin') {
+    if (this._platform === 'darwin') {
```
Comment (JoelEinbinder) [inline src/cli/injected/recorder.ts:377]: this is a drive-by? I assume the process.platform was being compiled by webpack into the platform of whoever compiled the bundle. Why not use navigator.platform?

Comment (pavelfeldman) [reply src/cli/injected/recorder.ts:377]: Yeah, i would also use navigator for this to feel more webby...

**Comment by pavelfeldman** [comment]
Can we get a test for it as well?

File: src/cli/injected/recorder.ts (hunk 2 above)
Comment (dgozman) [reply src/cli/injected/recorder.ts]: Done.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**